src/main/java/seedu/mindmymoney/command/UpdateCommand.java

```diff
@@ -140,6 +144,29 @@ public void updateExpenditure() throws MindMyMoneyException {
         }
     }
 
+    /**
+     * Checks if the fields in the update command is similar to the fields in the expenditure in the list.
+     *
+     * @param index index of expenditure to update.
+     * @param newPaymentMethod new payment method field to be updated.
+     * @param newCategory new category field to be updated.
+     * @param newDescription new description field to be updated.
+     * @param newAmountAsFloat new amount field to be updated.
+     * @param newTime new time field to be updated.
+     * @return true if fields are similar, false otherwise.
+     */
+    public boolean isSimilarExpenditure(int index, String newPaymentMethod, String newCategory, String newDescription,
+                             float newAmountAsFloat, String newTime) {
+        if (expenditureList.get(index).getPaymentMethod().equals(newPaymentMethod)
+                && expenditureList.get(index).getCategory().equals(newCategory)
+                && expenditureList.get(index).getDescription().equals(newDescription)
+                && expenditureList.get(index).getAmount() == newAmountAsFloat
+                && expenditureList.get(index).getTime().equals(newTime)) {
```

Comment:
Maybe can refactor out each condition to improve SLAP and readability

Reply:
resolved.